src/foam/nanos/auth/resetPassword/ResetPasswordTokenService.js

```diff
@@ -168,6 +174,30 @@ foam.CLASS({
         ((DAO) x.get("emailMessageDAO")).put(message);
         return true;
       `
+    },
+    {
+      name: 'getMenu',
```

Comment:
Can this and getTemplateName not be a function?
getParameter(String key, String default)

Reply:
Parameters can be null, so I need a null check, I find the function cleaner and easier to override. what specific issue with the function?

Reply:
What I mean is getMenu and getTemplateName don't need to be separate functions.  And since you are taking from a map - were you planning on a function for each possible key? 
I would prefer
getParameter(String key, String default)  - just one function.
the default can be provided or the model can have a Map property with defaults.

Reply:
Ok got you now.